core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogItemDo.java

```diff
@@ -65,6 +64,19 @@ public SubscriptionSupportInternal subscriptions() {
         throw new UnsupportedOperationException();
     }
     
+    /**
+     * @throws UnsupportedOperationException; relations are not supported for catalog items
+     */
+    @Override
+    public RelationSupportInternal<CatalogItem<T,SpecT>> relations() {
+        throw new UnsupportedOperationException();
```

Comment:
I think it's better to return an immutable empty RelationSupport object instead. This will allow to avoid special casing `CatalogItem` when working with `BrooklynObject`s.